types/index.d.ts

```diff
@@ -1703,6 +1703,12 @@ export function padCharsStart(
 export function padStart(targetLength: number, value: string): string;
 export function padStart(targetLength: number): (value: string) => string;
 
+/**
+ * Sort a list of objects by a property.
+ */
+export function sortByProp<T>(prop: keyof T | number, list: T[]): T[];
+export function sortByProp<T>(prop: keyof T | number): (list: T[]) => T[];
```

Comment:
Hey @char0n , I improve a little the types with generics.

Reply:
I did some final changes as well ;] Thanks